lampo-jsonrpc/src/lib.rs

```diff
@@ -118,184 +117,152 @@ impl<T: Send + Sync + 'static> JSONRPCv2<T> {
+                            return Err(err);
+                        }
+                        log::info!("Accepting connection: `{:?}`", accept);
+                        let stream = accept?.0;
+                        self.sources.register(
+                            RPCEvent::Connect,
+                            &stream,
+                            popol::interest::READ | popol::interest::WRITE,
+                        );
+                        self.open_streams.insert(stream.as_raw_fd(), stream);
+                        break;
+                    },
+                    RPCEvent::Connect if event.is_readable() => {
+                        self.read(&mut event)?;
                     }
-                    RPCEvent::Connect(addr) => {
-                        if event.is_hangup() {
-                            break;
+                    RPCEvent::Connect if event.is_writable() => {
+                        let stream = self.open_streams.get_mut(&event.as_raw_fd()).unwrap();
+                        match stream.flush() {
+                            // In this case, we've written all the data, we
+                            // are no longer interested in writing to this
+                            // socket.
+                            Ok(()) => {
+                                log::trace!("reading ended");
+                                event.source.unset(popol::interest::WRITE);
+                            }
+                            // In this case, the write couldn't complete. Set
+                            // our interest to `WRITE` to be notified when the
+                            // socket is ready to write again.
+                            Err(err)
+                                if [io::ErrorKind::WouldBlock, io::ErrorKind::WriteZero]
+                                    .contains(&err.kind()) =>
+                            {
+                                log::info!("reading return an error: {:?}", err);
+                                event.source.set(popol::interest::READ);
+                                break;
+                            }
+                            Err(err) => {
+                                return Err(err);
+                            }
                         }
-                        if event.is_error() {
-                            log::error!(target: "jsonrpc", "an error occurs");
+                    }
+                    RPCEvent::Connect => {
+                        if event.is_hangup() || event.is_error() {
+                            log::error!(target: "jsonrpc", "an error occurs: {:?}", event);
                             continue;
                         }
 
                         if event.is_invalid() {
-                            log::info!(target: "jsonrpc", "event invalid, unregister event from the tracking one");
+                            log::warn!(target: "jsonrpc", "event invalid: {:?}", event);
```

Comment:
This gives me the warning everytime I run `lampo-cli --network regtest getinfo`. 
```
2024-05-24T16:13:35.386Z WARN jsonrpc event invalid: Event { key: Connect, source: Source { fd: 15, events: 519, revents: 32 } }. [lampo-jsonrpc/src/lib.rs:281]
```
Should we take care of this or should we ignore this everytime.

This is the basic flow of events whenever I am querying the node. 
1. `RPCEvent::Accept`
2. `RPCEvent::Connect && event.is_readable()`
3. `RPCEvent::Connect and the event.is_invalid()` -> This throws the warning.